### DISCUSSION TITLE
Custom Java accessing the Mirth Truststore

### DISCUSSION THREAD

**Discussion by mcswaip1** [nextgenhealthcare/connect#5810] [Q&A] [answered]
This may be a really dumb question, but I need to access the mirth trust store to retrieve a certificate to use when connecting to RabbitMQ with TLS. But from what I can tell, the trust store for Mirth 4.3 is in the database. So how do I get access to that information to create the RabbitMQ TLS connection?
Thank you in advance.

**Comment by pacmano1** [comment]
Can you rephrase that? 

If you are connecting to a service with a self signed certificate you will need to add the certificate authority to the cacerts file for java on your server, not a mirth file.  That assumes you are not using the commercial SSL plugin. 

Mirth's own certificate is not in the database, it is in a keystore file in Mirth's app data directory.

**Reply by mcswaip1** [reply]
NOTE: We are using the Nextgen SSL Extension

Sorry. Let me give a bit of history. I took a ton of JavaScript and moved it into java to allow unit tests and reduce code duplication between channels.

Some of the JS code dealt with communicating with RabbitMQ. To simplify that code I directly accessed the amqp/rabbitmq objects/methods.

Now I have been tasked with adding TLS into the mix.
So, I have the certificate loaded into the trust store in Mirth, but I can't seem to find a way to access it so that I can create the SSLContext when I instantiate the ConnectionFactory for RabbitMQ.

Thank you

**Reply by pacmano1** [reply, marked as answer]
LOL, much better question.  

I'm not sure on that, I think some other folks have been looking for a
more seamless/simple way to access that cert store via API.  I'll defer here to other folks.

**Reply by pacmano1** [reply]
On a sub note, mirth has code templates, why was that not suitable to reduce duplicate code in channels?  I get the unit test thing of course. Just curious.

**Comment by jonbartels** [comment]
The NG SSL Extension stores its keystore in a JKS or JCEKS format base64 encoded in the `configuration` table.

One way to see it is to do a server export, within that export you'll see your key- and trust- store along with the passwords.

For your use case, do you need *A* trust store or *the* truststore that Mirth and the NG SSL extension are using?
